src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -220,8 +232,14 @@ public virtual async Task<T> GetConfigurationAsync(CancellationToken cancel)
 
             // If metadata exists return it.
             if (_currentConfiguration != null)
+            {
+                TrackGetConfiguration(
+                    methodStartTicks, IdentityModelTelemetryUtil.Requested, IdentityModelTelemetryUtil.Success, string.Empty);
```

Comment:
The logic that got us here does not necessarily imply that valid metadata was received.